### ISSUE TITLE
Research: static link libsecp256k1 

### ISSUE THREAD

**Issue by ashchan** [cryptape/ruby-bitcoin-secp256k1#6]
Embed C module (example: unicorn) instead of using external libsecp26bk1 using ffi.

**Comment by ashchan** [comment]
@classicalliu Any thoughts? Interested to give it a try?

**Comment by classicalliu** [comment]
@ashchan Means building native extensions while installing? I'll have a try

**Comment by ashchan** [comment]
> @ashchan Means building native extensions while installing? I'll have a try

Right. Such that the gem doesn't depend on the dynamic linked secp256k1 on the OS. @xxuejie is this understanding correct?
